### DISCUSSION TITLE
Cron automatisch starten?

### DISCUSSION THREAD

**Discussion by tops4u** [buanet/ioBroker.docker#483] [Questions & Answers / Fragen & Antworten] [answered]
Hallo zusammen,

ich benutze einen Cronjob um exportierte CSV Dateien auf einen Webserver in der DMZ zu pushen. Leider habe ich es bisher nicht hinbekommen, Cron automatisch beim Container Restart mitstarten zu lassen, so dass immer wieder keine Daten gepusht werden. 

Wie kann ich Cron automatisch mit dem Container restarten lassen? Bin leider ein absolutes Docker-Dummy... 

Ach ja mein Image ist 9.0.1 und ich werde das wohl auch so lassen, da ich ein paar lokale Dateien verändert habe. NodeJS und Co Update schaffe ich.

**Comment by buanet** [comment, marked as answer]
Hallo,
so ganz verstehe ich deinen Ansatz nicht. 
> Ach ja mein Image ist 9.0.1 und ich werde das wohl auch so lassen, da ich ein paar lokale Dateien verändert habe.

Wenn ich das höre, höre ich nur, dass du Docker noch nicht verstanden hast. Vielleicht lohnt sich für dich ein Blick in die Doku, Stichwort "Userscripts". Damit solltest du jede Manipulation im Image updatefest machen können. Aber das nur am Rande....

> Wie kann ich Cron automatisch mit dem Container restarten lassen? 

Nein. Keine Manipulation am Image. Das ist nicht Sinn der Sache!

> ich benutze einen Cronjob um exportierte CSV Dateien auf einen Webserver in der DMZ zu pushen. Leider habe ich es bisher nicht hinbekommen, Cron automatisch beim Container Restart mitstarten zu lassen

Wie pusht du denn die Dateien? Ohne das jetzt getestet zu haben, würde ich meinen, dass du das über den ioBroker/ javascript-Adapter machen kannst. Der übernimmt dann auch die Zeitplanung für dich. 

Wenn du das aus irgendeinem Grund nicht über ioBroker machen willst, dann mache es über einen separaten (Cron)-Container und arbeite mit Mounts. Das wäre der korrekte Docker-Ansatz.

MfG,
André

**Comment by tops4u** [comment]
Ja ich weiss, das habe ich nun auch an anderer Stelle schon gesehen. Was wäre denn der beste Weg per SCP CSV Dateien aus dem IoBroker Image an einen Webserver in der DMZ zu senden? Ich brauche dazu auch noch die SSH Keys, denn ich will keine Accountdaten in Scripts parken. 

SCP scheint aus JavaScript etwas tricky. Und was anderes möchte ich nicht aus meiner trusted zone in die DMZ.

Einen Zoo von Containern möchte ich auch nicht unbedingt haben.

**Reply by buanet** [reply]
> Was wäre denn der beste Weg per SCP CSV Dateien aus dem IoBroker Image an einen Webserver in der DMZ zu senden?

Mir ist nicht so ganz klar was du hier jetzt erwartest. Die grundsätzlichen Ansätze habe ich dir geschildert.
Meines Erachtens wäre die Option über Javascript der beste Weg. Du kannst von dort aus auch einfach nur ein Shell-Script aufrufen. Als Speicherort würde ich ein Verzeichnis Mount, analog zum /opt/iobroker mount, empfehlen. Dann einfach dein Script da rein, im Javascript Adapter [exec](https://www.iobroker.net/docu/index-81.htm?page_id=5809&lang=en#exec_8211_execute_some_OS_command_like_8220cp_file1_file28221) erlauben und los geht's.

> Ich brauche dazu auch noch die SSH Keys, denn ich will keine Accountdaten in Scripts parken.

Musst du dann wohl auch für die Einrichtung in ein Shellscript/ Userscript scripten. Für die Keys selbst nutzt du ein Mount.

> SCP scheint aus JavaScript etwas tricky. 

Kann ich nicht bewerten. Aber du musst scp ja nicht in Javascript machen, sondern in der Shell. Nur der Aufruf des Shellscripts erfolgt dann über Javascript...

> Einen Zoo von Containern möchte ich auch nicht unbedingt haben.

Dann ist Docker für dich vielleicht nicht das Richtige und du suchst in Wirklichkeit nach einer virtuellen Maschine. Denn genau dieser "Container-Zoo" ist ja die Idee/ Definition der Containerisierungs-Technik. Jeder Service erhält seinen eigenen Container. Aber das führt hier jetzt zu weit...

MfG,
André
